File: PosApi/Models/Categoria.cs
namespace PosApi.Models;

public class Categoria
{
    public int IdCategoria { get; set; }
    public string Nombre { get; set; } = null!;
    public bool Activo { get; set; }
    public DateTime FechaRegistro { get; set; }

    public List<Producto> Productos { get; set; } = new();
}

File: lib/models/categoria.dart
// lib/models/categoria.dart

class Categoria {
  final int idCategoria;
  final String nombre;
  final bool activo;

  Categoria({
    required this.idCategoria,
    required this.nombre,
    required this.activo,
  });

  factory Categoria.fromJson(Map<String, dynamic> json) {
    return Categoria(
      idCategoria: (json['idCategoria'] ?? json['IdCategoria']) as int,
      nombre: (json['nombre'] ?? json['Nombre']) as String,
      activo: (json['activo'] ?? json['Activo'] ?? 1) is bool
          ? (json['activo'] ?? json['Activo']) as bool
          : ((json['activo'] ?? json['Activo'] ?? 1) as num) == 1,
    );
  }
}
